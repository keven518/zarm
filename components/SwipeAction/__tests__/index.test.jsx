import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SwipeAction from '../index';

describe('SwipeAction', () => {
  it('renders correctly', () => {
    jest.useFakeTimers();
    const wrapper = mount(
      <SwipeAction
        left={[
          {
            theme: 'info',
            text: '左按钮1',
            onClick: jest.fn(),
          },
          {
            theme: 'warning',
            text: '左按钮2',
            onClick: jest.fn(),
          },
        ]}
        right={[
          {
            theme: 'error',
            text: '右按钮1',
            onClick: jest.fn(),
          },
          {
            theme: 'success',
            text: '右按钮2',
            onClick: jest.fn(),
          },
        ]}>
        <div>左右都能滑动</div>
      </SwipeAction>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('left only', () => {
    const wrapper = render(
      <SwipeAction
        left={[
          {
            theme: 'info',
            text: '左按钮1',
            onClick: jest.fn(),
          },
          {
            theme: 'warning',
            text: '左按钮2',
            onClick: jest.fn(),
          },
        ]}>
        <div>右滑</div>
      </SwipeAction>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('right only', () => {
    const wrapper = render(
      <SwipeAction
        right={[
          {
            theme: 'info',
            text: '左按钮1',
            onClick: jest.fn(),
          },
          {
            theme: 'warning',
            text: '左按钮2',
            onClick: jest.fn(),
          },
        ]}>
        <div>左滑</div>
      </SwipeAction>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
