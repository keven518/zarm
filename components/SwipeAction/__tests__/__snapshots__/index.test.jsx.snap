// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeAction left only 1`] = `
<div
  class="za-swipeaction"
>
  <div
    class="za-swipeaction-actions-left"
  >
    <div
      class="za-swipeaction-button theme-info"
    >
      <div
        class="za-swipeaction-text"
      >
        左按钮1
      </div>
    </div>
    <div
      class="za-swipeaction-button theme-warning"
    >
      <div
        class="za-swipeaction-text"
      >
        左按钮2
      </div>
    </div>
  </div>
  <div
    class="za-swipeaction-content"
    style="-webkit-transition-duration:undefinedms;transition-duration:undefinedms;-webkit-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  >
    <div>
      右滑
    </div>
  </div>
</div>
`;

exports[`SwipeAction renders correctly 1`] = `
<SwipeAction
  animationDuration={300}
  left={
    Array [
      Object {
        "onClick": [Function],
        "text": "左按钮1",
        "theme": "info",
      },
      Object {
        "onClick": [Function],
        "text": "左按钮2",
        "theme": "warning",
      },
    ]
  }
  moveDistanceRatio={0.5}
  moveTimeSpan={300}
  offset={10}
  prefixCls="za-swipeaction"
  right={
    Array [
      Object {
        "onClick": [Function],
        "text": "右按钮1",
        "theme": "error",
      },
      Object {
        "onClick": [Function],
        "text": "右按钮2",
        "theme": "success",
      },
    ]
  }
>
  <div
    className="za-swipeaction"
  >
    <div
      className="za-swipeaction-actions-left"
    >
      <div
        className="za-swipeaction-button theme-info"
        onClick={[Function]}
      >
        <div
          className="za-swipeaction-text"
        >
          左按钮1
        </div>
      </div>
      <div
        className="za-swipeaction-button theme-warning"
        onClick={[Function]}
      >
        <div
          className="za-swipeaction-text"
        >
          左按钮2
        </div>
      </div>
    </div>
    <div
      className="za-swipeaction-actions-right"
    >
      <div
        className="za-swipeaction-button theme-error"
        onClick={[Function]}
      >
        <div
          className="za-swipeaction-text"
        >
          右按钮1
        </div>
      </div>
      <div
        className="za-swipeaction-button theme-success"
        onClick={[Function]}
      >
        <div
          className="za-swipeaction-text"
        >
          右按钮2
        </div>
      </div>
    </div>
    <Drag
      onDragEnd={[Function]}
      onDragMove={[Function]}
      onDragStart={[Function]}
    >
      <div
        className="za-swipeaction-content"
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0, 0)",
            "WebkitTransitionDuration": "undefinedms",
            "transform": "translate3d(0px, 0, 0)",
            "transitionDuration": "undefinedms",
          }
        }
      >
        <div>
          左右都能滑动
        </div>
      </div>
    </Drag>
  </div>
</SwipeAction>
`;

exports[`SwipeAction right only 1`] = `
<div
  class="za-swipeaction"
>
  <div
    class="za-swipeaction-actions-right"
  >
    <div
      class="za-swipeaction-button theme-info"
    >
      <div
        class="za-swipeaction-text"
      >
        左按钮1
      </div>
    </div>
    <div
      class="za-swipeaction-button theme-warning"
    >
      <div
        class="za-swipeaction-text"
      >
        左按钮2
      </div>
    </div>
  </div>
  <div
    class="za-swipeaction-content"
    style="-webkit-transition-duration:undefinedms;transition-duration:undefinedms;-webkit-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  >
    <div>
      左滑
    </div>
  </div>
</div>
`;
